src/main/java/cn/nukkit/block/BlockBed.java

```diff
@@ -124,39 +122,63 @@ public boolean onActivate(@Nonnull Item item) {
     @Override
     public boolean onActivate(@Nonnull Item item, Player player) {
 
-        if (!(this.level.getDimension() == Level.DIMENSION_OVERWORLD)) {
-            CompoundTag tag = EntityPrimedTNT.getDefaultNBT(this).putShort("Fuse", 0);
-            new EntityPrimedTNT(this.level.getChunk(this.getFloorX() >> 4, this.getFloorZ() >> 4), tag);
-            return true;
-        }
-
         BlockFace dir = getBlockFace();
+
+        boolean shouldExplode = this.level.getDimension() == Level.DIMENSION_OVERWORLD;
```

Comment:
You accidentally mistook the Respawn Anchor for the bed. The bed is to explode in other dimensions and the respawn anchor in the overworld.

Reply:
Thank you, imagine if it got into production, the reaction of the people after trying to sleep at night haha